server/src/main/java/org/elasticsearch/common/lucene/search/function/ScriptScoreFunction.java

```diff
@@ -84,10 +86,15 @@ public Explanation explainScore(int docId, Explanation subQueryScore) throws IOE
                     exp = ((ExplainableScoreScript) leafScript).explain(subQueryScore);
                 } else {
                     double score = score(docId, subQueryScore.getValue().floatValue());
+                    Explanation customExplanation = explanation.get(subQueryScore.getValue().floatValue(), null);
                     // info about params already included in sScript
                     String explanation = "script score function, computed with script:\"" + sScript + "\"";
-                    Explanation scoreExp = Explanation.match(subQueryScore.getValue(), "_score: ", subQueryScore);
-                    return Explanation.match((float) score, explanation, scoreExp);
+                    if (customExplanation != null) {
+                        return Explanation.match((float) score, explanation, customExplanation);
+                    } else {
+                        Explanation scoreExp = Explanation.match(subQueryScore.getValue(), "_score: ", subQueryScore);
+                        return Explanation.match((float) score, explanation, scoreExp);
```

Comment:
We should still include the sub-query explanation even if a custom one is provided. We should consider the custom one as a replacement for `"script score function, computed with script..."`, but not the entire thing.

The reason being, is that the sub-query is vitally important to functionscore queries.

Reply:
That makes sense.  Here's an example of what that would look like.  Let me know if that's still not quite what you were thinking.  I'll update the description as well.  Easy change.  

```
"value" : 3.0,
"description" : "min of:",
"details" : [
  {
    "value" : 3.0,
    "description" : "An example optional custom description to explain details for this script's execution; we'll provide a default one if you leave this out.",
    "details" : [
      {
        "value" : 0.18952844,
        "description" : "_score: ",
        "details" : [
          {
```

Reply:
yes, very good